### PR TITLE
[GLUTEN-3559][VL] Fix UTs failing in spark 3.4 which have assertions for SMJ node

### PR DESCRIPTION
**Pull request by ayushi-agarwal**
## What changes were proposed in this pull request?

This PR fixes the following suites:

1. GlutenJoinSuite
2. GlutenKeyGroupedPartitioningSuite
3. GlutenWriteDistributionAndOrderingSuite

(Partially Fixes: \#3559)

## How was this patch tested?
Ran UTs locally

### CODE REVIEW

**Comment by ayushi-agarwal** [comment]
cc: @JkSelf

**Comment by ayushi-agarwal** [comment]
@JkSelf @ulysses-you @zhli1142015 Could you please help in reviewing this PR?

File: gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 1]
```diff
@@ -68,44 +68,14 @@ class VeloxTestSettings extends BackendTestSettings {
     .exclude("partitioned join: number of buckets mismatch should trigger shuffle")
     .exclude("partitioned join: only one side reports partitioning")
     .exclude("partitioned join: join with two partition keys and different # of partition keys")
+    // disable as both checks for SMJ node
     .excludeByPrefix("SPARK-41413: partitioned join:")
-    .exclude("SPARK-42038: partially clustered: with different partition keys and both sides partially clustered")
```
[the remark this answers is not in the document]
Comment (ayushi-agarwal) [reply gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:72]: Yes, they didn't require any fix and were passing so I am just enabling them in this PR.

**Review by zhli1142015 — APPROVED** [review]
LGTM, thanks

File: gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 2]
```diff
@@ -68,44 +68,14 @@ class VeloxTestSettings extends BackendTestSettings {
...
   enableSuite[GlutenMetadataColumnSuite]
   enableSuite[GlutenSupportsCatalogOptionsSuite]
   enableSuite[GlutenTableCapabilityCheckSuite]
   enableSuite[GlutenWriteDistributionAndOrderingSuite]
-    .exclude("ordered distribution and sort with same exprs: append")
-    .exclude("ordered distribution and sort with same exprs: overwrite")
-    .exclude("ordered distribution and sort with same exprs: overwriteDynamic")
-    .exclude("clustered distribution and sort with same exprs: append")
-    .exclude("clustered distribution and sort with same exprs: overwrite")
-    .exclude("clustered distribution and sort with same exprs: overwriteDynamic")
```
Comment (ayushi-agarwal) [inline gluten-ut/spark34/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:89]: @ulysses-you These (GlutenWriteDistributionAndOrderingSuite) are the one which were passing without any changes, so I just removed them from exclude

**Review by JkSelf — APPROVED** [review]
LGTM.

**Review by ulysses-you — APPROVED** [review]

**Comment by ayushi-agarwal** [comment]
Thank you @JkSelf @ulysses-you @zhli1142015

**Pull request merged**